rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetTelemetryNodeTest.java

```diff
@@ -15,72 +15,341 @@
+    }
+
+    @Test
+    public void givenUseMetadataIntervalPatternsIsTrueAndFetchModeIsLastAndAggregationIsMax_whenOnMsg_thenTellSuccess() throws TbNodeException {
+        config.setLatestTsKeyNames(List.of("temperature", "humidity"));
+        config.setUseMetadataIntervalPatterns(true);
+        config.setStartIntervalPattern("${mdStartInterval}");
+        config.setEndIntervalPattern("$[msgEndInterval]");
+        config.setFetchMode("LAST");
+        config.setAggregation("MAX");
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        mockTimeseriesService();
+        long ts = System.currentTimeMillis();
+        List<TsKvEntry> tsKvEntries = List.of(
+                new BasicTsKvEntry(ts - 5, new DoubleDataEntry("temperature", 22.4)),
+                new BasicTsKvEntry(ts - 4, new DoubleDataEntry("humidity", 55.5))
+        );
+        given(timeseriesServiceMock.findAll(any(TenantId.class), any(EntityId.class), anyList())).willReturn(Futures.immediateFuture(tsKvEntries));
+
+        long startTs = 1719220350000L;
+        long endTs = 1719220353000L;
+        TbMsgMetaData metaData = new TbMsgMetaData();
+        metaData.putValue("mdStartInterval", String.valueOf(startTs));
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, DEVICE_ID, metaData, "{\"msgEndInterval\":\"" + endTs + "\"}");
+        node.onMsg(ctxMock, msg);
+
+        List<ReadTsKvQuery> expectedReadTsKvQueryList = List.of(
+                new BaseReadTsKvQuery("temperature", startTs, endTs, 1, 1, Aggregation.NONE, "DESC"),
+                new BaseReadTsKvQuery("humidity", startTs, endTs, 1, 1, Aggregation.NONE, "DESC")
+        );
+        verifyReadTsKvQueryList(expectedReadTsKvQueryList, false);
+        verifyOutgoingMsg(msg);
+    }
+
+    @Test
+    public void givenUseMetadataIntervalPatternsIsFalseAndTsKeyNamesPatternsAndFetchModeIsFirst_whenOnMsg_thenTellFailure() throws TbNodeException {
+        config.setLatestTsKeyNames(List.of("temperature", "${mdTsKey}", "$[msgTsKey]"));;
+        config.setFetchMode("FIRST");
+        config.setStartInterval(6);
+        config.setEndInterval(1);
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        mockTimeseriesService();
+        String errorMsg = "Something went wrong";
+        given(timeseriesServiceMock.findAll(any(TenantId.class), any(EntityId.class), anyList())).willReturn(Futures.immediateFailedFuture(new RuntimeException(errorMsg)));
+
+        TbMsgMetaData metaData = new TbMsgMetaData();
+        metaData.putValue("mdTsKey", "humidity");
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, DEVICE_ID, metaData, "{\"msgTsKey\":\"pressure\"}");
+        node.onMsg(ctxMock, msg);
+
+        long ts = System.currentTimeMillis();
+        long startTs = ts - TimeUnit.SECONDS.toMillis(config.getStartInterval());
+        long endTs = ts - TimeUnit.SECONDS.toMillis(config.getEndInterval());
```

Comment:
Why seconds? Looks like you are using default config for time units and it should be minutes